### ISSUE TITLE
global template ignored

### ISSUE THREAD

**Issue by juanolon** [zk-org/zk#394]
### Check if applicable

- [X] I have searched the existing issues (**required**)
- [ ] I'm willing to help fix the problem and contribute a pull request

### Describe the bug

Hello,
when i make a new note with `zk new` the [note].template entry in the config file is being ignored. i have the same entry on [group.custom.note].template for testing, and then works with `zk new --group custom`.

### How to reproduce?

1. `zk new`
2. expected template is not used

### zk configuration

```toml
[notebook]
dir = "~/notes"

[note]
language = "en"
# Template used to generate a note's filename, without extension.
filename = "{{id}}-{{format-date now '%Y-%m-%d'}}"

# The file extension used for the notes.
extension = "md"

template = "hugo.md"

# GROUP OVERRIDES
[group.custom.note]
template = "hugo.md"
```

### Environment

```bash
zk 0.14.0
system: Linux 6.1.0-18-amd64 x86_64 GNU/Linux
```

**Comment by tjex** [comment]
when running `zk new`, what note template do you end up with? Is it still the default.md template?

Have you tried running `zk index` or quitting and re-running zk? Perhaps zk still has an old config loaded, where `[note]` settings are set to default.md as the template.

**Comment by juanolon** [comment]
Hi,
yes, the note is created using the default template I guess, well, the note is empty.
I did kill all zk instances and redo the index as proposed, but still the same.

**Comment by tjex** [comment]
Can you try with the default config including the default title for `[note]`:

```
[note]
default-title = "Untitled"
template = "default.md"
```

See if that generates a blank note with title=Untitled when running `zk new`.

If it does, swap out `template = "default.md"` with `... "hugo.md"` and see what happens. 

Do you have multiple zk repos and / or are your running these commands from within the zk repo root folder?

Because, if you're running the command from outside the zk wiki's root folder, you need to have the `NOTEBOOK_DIR` env variable set, otherwise `zk` won't know where to look for the config file etc, [see docs](https://github.com/zk-org/zk/blob/main/docs/notebook.md)

**Comment by tjex** [comment]
I tested by making a new folder at `~/notes` and using only the config you have above. I disabled my shell notebook dir env variable.
I get this error when running `zk new` from `~/notes`:

`zk: error: failed to open notebook: failed to read config: notebook.dir should not be set on local configuration`

I removed the `[notebook]` section, and it works fine. Here is the tree if it helps:

```tree
.
├── .zk
│   ├── config.toml
│   ├── notebook.db
│   └── templates
│       ├── default.md
│       └── hugo.md
└── oxk8-2024-03-18.md
```
